addons/account_debit_note/views/account_move_view.xml

```diff
@@ -32,4 +32,40 @@ if records:
         </field>
     </record>
 
+    <record id="view_account_move_filter_debit" model="ir.ui.view">
+        <field name="name">account.move.filter.debit</field>
+        <field name="model">account.move</field>
+        <field name="inherit_id" ref="account.view_account_move_filter" />
+        <field name="arch" type="xml">
+            <filter name="reversed" position="after">
+                <filter string="Debit Note" name="debit_note_filter"
+                    domain="[('debit_origin_id', '!=', False)]" />
+            </filter>
+        </field>
+    </record>
+
+    <record id="view_account_invoice_filter_debit" model="ir.ui.view">
+        <field name="name">account.invoice.select.debit</field>
+        <field name="model">account.move</field>
+        <field name="inherit_id" ref="account.view_account_invoice_filter" />
+        <field name="arch" type="xml">
+            <filter name="to_check" position="before">
+                <separator />
+                <filter string="Debit Note" name="debit_note_filter"
+                    domain="[('debit_origin_id', '!=', False)]" />
```

Comment:
Since you add before a filter, the separator has to be after
```suggestion
                <filter string="Debit Note" name="debit_note_filter"
                    domain="[('debit_origin_id', '!=', False)]" />
                <separator />
```